larreco/RecoAlg/PMAlg/PmaElement3D.cxx

```diff
@@ -0,0 +1,342 @@
+/**
+ *  @file   PmaElement3D.cxx
+ *
+ *  @author D.Stefan and R.Sulej
+ * 
+ *  @brief  Implementation of the Projection Matching Algorithm
+ *
+ *          Base for 3D segments and track nodes. See PmaTrack3D.h file for details.
+ *          See PmaTrack3D.h file for details.
+ */
+
+#include "larreco/RecoAlg/PMAlg/PmaElement3D.h"
+#include "larreco/RecoAlg/PMAlg/SortedObjects.h"
+#include "larreco/RecoAlg/PMAlg/Utilities.h"
+
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+// Impact factors on the objective function:  U     V     Z
+float pma::Element3D::fOptFactors[3] =     { 0.2F, 0.8F, 1.0F };
+
+pma::Element3D::Element3D() :
+	fTPC(-1), fCryo(-1),
+	fFrozen(false),
+	fHitsRadius(0)
+{
+	fNThisHitsEnabledAll = 0;
+	for (unsigned int i = 0; i < 3; i++)
```

Comment:
Instead of using `3` explicitly, it would help to create a symbolic constant set to 3 and then use it for the array sizes as well as to iterate over the elements.
Another option to consider is to switch to `std::array<>` since it would initialize the values to 0 and allows standard iteration.